pilot/pkg/model/notifier.go

```diff
@@ -0,0 +1,57 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package model
+
+import (
+	"sync"
+
+	"go.uber.org/atomic"
+)
+
+type ActiveNotifier struct {
+	state        *atomic.Bool
+	handlers     []func(state bool)
+	handlerMutex sync.Mutex
+}
+
+func (n *ActiveNotifier) Current() bool {
```

Comment:
consider `CurrentlyActive()`. This seemed fine when I read the notifier itself but feels awkward later when used in the code